packages/orm/src/relations/CustomCollection.ts

```diff
@@ -14,14 +15,22 @@ export type CustomCollectionOpts<T extends Entity, U extends Entity> = {
   add?: (entity: T, other: U) => void;
   remove?: (entity: T, other: U) => void;
   /** Whether the reference is loaded, even w/o an explicit `.load` call, i.e. for DeepNew test instances. */
-  isLoaded: () => boolean;
+  isLoaded: (entity: T) => boolean;
   /**
    * A load hint that can be used to speculatively traverse the dependencies of this collection.  Currently
    * only used by em.importEntity and can be omitted if the collection isn't used with it.
    */
   loadHint?: ((entity: T) => LoadHint<T>) | LoadHint<T>;
 };
 
+export function hasCustomCollection<T extends Entity, U extends Entity>(
+  opts: CustomCollectionOpts<T, U>,
+): CustomCollection<T, U> {
```

Comment:
The `readonly coll = new CustomCollection` callers now need to be `coll = hasCustomCollection`